### PR TITLE
DM-1174: fix streaming docs

### PR DESCRIPTION
**Pull request by mattclarke**
### Description of work
Some of the documentation was out of date, specifically relating to streams.
This PR fixes that and updates other parts of the commands.md file too.

Also, removed some legacy documentation.

### Issue

Closes DM-1174

### Acceptance Criteria

Make sure the commands.md file makes sense.

### Unit Tests

N/A

### Other

N/A

---

## Code Review (To be filled in by the reviewer only)

- [ ] Is the code of an acceptable quality?
- [ ] Do the changes function as described and is it robust?

---

## Nominate for Group Code Review (Anyone can nominate it)
Indicate if you think the code should be reviewed in a Thursday code review session.

- [ ] Recommend for group code review

Also, nominate it on the code_review Slack channel (does someone want to automate this?).

### CODE REVIEW

File: documentation/commands.md [hunk 1]
```diff
@@ -1,233 +1,243 @@
...
-In the command, the `nexus_structure` defines the HDF hierarchy.
-The `nexus_structure` represents the HDF root object.  The following example
-shows how the HDF tree can be constructed using `children`.
-A child of type `stream` is a marker that a `HDFWriterModule` will insert
-the data into from a Kafka stream at that point in the hierarchy.  The options under
-the key `stream` specify the stream details containing, at least, the topic, the source name and
-the `HDFWriterModule` which should be used for writing.
-Depending on the `HDFWriterModule`, there will be more options specific to the
-`HDFWriterModule`.
+Commands are generally send through Kafka via the broker and topic specified by the
```
Comment (SkyToGround) [inline documentation/commands.md]:
```suggestion
Commands are generally sent through Kafka via the broker and topic specified by the
```

File: documentation/commands.md [hunk 2]
```diff
@@ -1,233 +1,243 @@
...
 
-- [Groups](docs/groups.md)
-- [~~Datasets~~ documentation not yet written]()
-- [Attributes](docs/attributes.md)
-- [~~File Attributes~~ documentation not yet written]()
-- [~~Streams~~ documentation not yet written]()
+- cmd: The command name, must be `FileWriter_new`.
+- job_id: A unique identifier for the request, a UUID for example.
+- broker: The Kafka broker to use for data.
+- start_time: The start time in milliseconds (UNIX epoch) for data to be written from. Optional, if not supplied then the Kafka message time is used.
```
Comment (SkyToGround) [inline documentation/commands.md]:
```suggestion
- start_time: The start time in milliseconds (UNIX epoch) for data to be written from. Optional, if not supplied then the timestamp of the Kafka message containing the start command is used.
```
Note: How does this work if the start command is supplied via JSON-file?

File: documentation/commands.md [hunk 3]
```diff
@@ -1,233 +1,243 @@
...
+- stop_time: The stop time in milliseconds (UNIX epoch) for data writing to stop. Optional, if not supplied then file writing continues until a stop command is received
+- service_id: The identifier for the instance of the file-writer that should handle this command. Optional, only needed if multiple file-writers present.
+- abort_on_uninitialised_stream: Whether to abort if the stream cannot be initialised. Optional, default is not to abort but carry on.
```
Comment (SkyToGround) [inline documentation/commands.md]: What does "cannot be initialised" mean?

File: documentation/commands.md [hunk 4]
```diff
@@ -1,233 +1,243 @@
...
-  "service_id": "[OPTIONAL] the_name_of_the_instance_which_should_interpret_this_command"
-}
-```
+In the example above, a group called my_test_group is defined which in turn has a stream and an attribute defined.
+The stream definition instructs the file-writer that there is data available which should be written to the containing group.
+The parameters for the stream definition are:
 
-#### Command to stop a file being written
+- writer_module: The FlatBuffers schema in the [streaming-data-types](https://github.com/ess-dmsc/streaming-data-types) repository that was used to serialise the data.
+- type: The type of the data. The possible types are defined in the schema declared in the `writer_module`.
```
Comment (SkyToGround) [inline documentation/commands.md]: We should have some information on the different requirements of `type` for the different writer modules but maybe that is outside the scope of this PR.

File: documentation/commands.md [hunk 5]
```diff
@@ -1,233 +1,243 @@
...
 
-```json
-{
-  "cmd": "FileWriter_stop",
-  "job_id": "job-unique-identifier",
-  "stop_time" : "[OPTIONAL] timestamp (int) in milliseconds",
-  "service_id": "[OPTIONAL] the_name_of_the_instance_which_should_interpret_this_command"
-}
-```
+Note: streams are automatically assigned a NeXus class based on the `writer_module`. [NEEDS TO BE CHECKED]
```
Comment (SkyToGround) [inline documentation/commands.md]: No need to check, some writer modules automatically assigns NeXus class, some does not. See also [DM-1220](https://jira.esss.lu.se/browse/DM-1220?jql=project%20%3D%2016000%20AND%20labels%20%3D%20%22FW%26FW%22).

File: documentation/commands.md (hunk 3 above)
Comment (matthew-d-jones) [reply documentation/commands.md]: :+1: It would be good to elaborate on that. Specifying an inaccessible broker in a stream URL, or a non-existent writer module are a couple of things which can cause this I think. There may be other possibilities.

File: documentation/commands.md (hunk 5 above)
Comment (matthew-d-jones) [reply documentation/commands.md]: There may even be multiple tickets about this because I think I created one on github at some stage.

File: documentation/commands.md (hunk 2 above)
Comment (mattclarke) [reply documentation/commands.md]: It uses the system clock

File: documentation/commands.md (hunk 5 above)
Comment (mattclarke) [reply documentation/commands.md]: Remind me to nominate it for next time.

File: documentation/commands.md (hunk 4 above)
Comment (mattclarke) [reply documentation/commands.md]: Agreed on both counts

File: documentation/commands.md (hunk 3 above)
Comment (mattclarke) [reply documentation/commands.md]: Basically anything that could go wrong: mistakes in the JSON, missing keys, no broker, multiple HDF issues...

File: documentation/commands.md [hunk 6]
```diff
@@ -241,3 +251,37 @@ Also:
...
+"attributes": {
+  "units": "Kelvin",
+  "error": 0.02
+}
+```
+Where possible the file-writer will try to infer the type of the data, in this example: a float.
+
+For strings and arrays it is necessary to specify the type, for example:
+
+```json
```
Comment (SkyToGround) [inline documentation/commands.md]: One final question: Do you know if it is possible to (somehow) use key-value pairs as well as arrays? If it is, how do you do that?

File: documentation/commands.md [hunk 7]
```diff
@@ -241,3 +251,37 @@ Also:
...
+"attributes": {
+  "units": "Kelvin",
+  "error": 0.02
+}
+```
+Where possible the file-writer will try to infer the type of the data, in this example: a float.
+
+For strings and arrays it is necessary to specify the type, for example:
```
Comment (SkyToGround) [inline documentation/commands.md]:
```suggestion
If more complicated types (i.e. arrays and strings)  are required, all attributes for a given dataset or group must be set using a long form version of the attributes.  In this version, each attribute is described using a set of  key-value pairs in an attribute specific dictionary. This dictionary is contained in a list of (possibly) many dictionaries . Example:
```
